### ISSUE TITLE
Old Code Standards: Java 1.4 Spring 1.3

### ISSUE THREAD

**Issue by techlogix** [swagger-api/swagger-codegen#4741]
##### Description

The code generated for Java Spring is 

The Code Gen for Java Spring needs to be updated. Currently it supports Java 1.7 and Spring 1.3.

It should be upgraded to support new dependencies in the POM - so that is supports and generates modern Java 9 and Spring 5 with the latest dependencies for SpringFox.

##### Steps to reproduce

Select Java Spring and inspect generated code stub.

**Comment by techlogix** [comment]
<properties>

<java.version>1.7</java.version>

<maven.compiler.source>${java.version}</maven.compiler.source>

<maven.compiler.target>${java.version}</maven.compiler.target>

<springfox-version>2.5.0</springfox-version>

</properties>


<parent>

<groupId>org.springframework.boot</groupId>

<artifactId>spring-boot-starter-parent</artifactId>

<version>1.3.5.RELEASE</version>

</parent>

**Comment by cbornet** [comment]
Well, neither Java 9 nor Spring 5 are released so that would be a bit early...
You can already use Java 8 with an option (-Djava8=true)

As for upgrading spring-boot and springfox dependencies, would you do a PR ?

**Comment by wing328** [comment]
@techlogix if you've time, we would welcome your enhancements to support Spring 5 via an additional option (e..g `useJavaSpring5`). Let me know if you need some good starting points.

**Comment by techlogix** [comment]
Yep - I can work on the enhancement. Pls msg me with some good starting points please.

**Comment by wing328** [comment]
@techlogix you will need to update the mustache templates for `spring-boot`, e.g. [pom.mustache](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/JavaSpring/libraries/spring-boot/pom.mustache#L9). You can find other templates in that folder or [the root folder or all spring templates](https://github.com/swagger-api/swagger-codegen/tree/master/modules/swagger-codegen/src/main/resources/JavaSpring)

To add another option (e.g. `useJavaSpring5`), https://github.com/swagger-api/swagger-codegen/pull/3112 is a good starting point.

Hope these help.
